lib/internal/worker.js

```diff
@@ -445,6 +445,10 @@ function setupChild(evalScript) {
 
   function fatalException(error) {
     debug(`[${threadId}] gets fatal exception`);
+
+    // set exit code for local worker's 'exit' listeners
+    process.exitCode = 1;
```

Comment:
That makes sense, but we should probably consider setting the code before calling handler (in both here and for process), to avoid current situation where it emits code 1 but `process.exitCode` has older value or undefined?

Reply:
@lundibundi I think that makes sense, yes. But since it affects all Node.js code, not just Workers, we probably should do it in a separate (and maybe even semver-major) PR.

The code you’d probably want to look at – if you want to tackle it yourself – is in `process._fatalException` in `lib/internal/bootstrap/node.js` and `FatalException()` in `src/node.cc`. We currently have hardcoded values for exiting in the latter function, so I think setting `process.exitCode` from `exit` handlers doesn’t have any effect at the moment?

Reply:
@addaleax Yeah, while looking into this one I looked though those functions (I actually first thought that exit code should be set on c++ side because we are listening on an exit callback from it).
From what I've seen both workers and process use node.cc/EmitExit which actually reevaluates `exitCode` after the event, so I think it will change the code. But indeed for the uncaught exception this is not the case, as it will always exit with 1 from `FatalException()` if it wasn't handled (or other hardcoded codes).
Should I open an issue first to track it or it should be okay to just submit PR?